code/game/machinery/Sleeper.dm

```diff
@@ -642,17 +624,19 @@
 				qdel(occupant)
 				occupant = null
 				var/obj/effect/decal/cleanable/ash/ashed = new /obj/effect/decal/cleanable/ash(loc)
-				ashed.layer = src.layer + 0.01
+				ashed.layer = layer + 0.01
 		playsound(src, 'sound/machines/ding.ogg', 50, 1)
-		on = 0
+		on = FALSE
 		if(occupant)
 			if(ishuman(occupant))
 				var/mob/living/carbon/human/H = occupant
-				if(isdiona(H))
+				if(isdiona(H)) 
 					if(H.h_style != "Popped Hair")
 						to_chat(H, "<span class = 'notice'>Your head pops!</span>")
 						playsound(src, 'sound/effects/pop.ogg', 50, 1)
 						H.h_style = "Popped Hair"
 						H.update_hair()
+				else if(isjusthuman(H) && Holiday == "April Fool's Day")
```

Comment:
I think the holidays themselves should be changed to defines instead of strings, both here and in the actual `Holidays.dm` file.